externs/plugin.externs.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @fileoverview Closure compiler externs for plugins. Use this file to define types that need to be left uncompiled to
+ *               avoid bracket notation and enforce type checking.
+ *
+ * @externs
+ */
+
+
+/**
+ * @type {Object}
+ */
+var pluginx;
+
+
+/**
+ * Namespace.
+ * @type {Object}
+ */
+pluginx.areadata;
+
+
+/**
+ * @typedef {{
+ *  label: (string|undefined),
+ *  text: (string|undefined),
+ *  icon: (string|undefined)
+ * }}
+ */
+pluginx.areadata.ListUIOptions;
+
+
+/**
+ * @typedef {{
+ *  id: (!string),
+ *  type: (plugin.areadata.AreaImportType|undefined),
```

Comment:
I got to talking with @schmidtk and he makes the argument that nothing in an extern should ever reference an internally defined type. This will eventually break with the upcoming module shift as these will no longer be referenceable here.

This is a pattern we've used elsewhere, but we should avoid going forward.

Reply:
This and MenuItemOptions are switched over to Object<string, *> for now.